controllers/component_build_controller.go

```diff
@@ -75,35 +75,78 @@ const (
 )
 
 var (
-	simpleBuildPipelineCreationTimeMetric       prometheus.Histogram
-	pipelinesAsCodeComponentProvisionTimeMetric prometheus.Histogram
+	initialComponentCreationBuildOrPaCProvisionTimeMetric prometheus.Histogram
+	simpleBuildPipelineCreationTimeMetric                 prometheus.Histogram
+	pipelinesAsCodeComponentProvisionTimeMetric           prometheus.Histogram
+	pipelinesAsCodeComponentUnconfigureTimeMetric         prometheus.Histogram
+	pipelinesAsCodeComponentTriggerTimeMetric             prometheus.Histogram
+	componentTimesForMetrics                                  = map[string]componentMetricsInfo{}
 )
 
+type componentMetricsInfo struct {
+	startTimestamp time.Time
+	requestType    string
+}
+
 func initMetrics() error {
 	buckets := getProvisionTimeMetricsBuckets()
 
+	initialComponentCreationBuildOrPaCProvisionTimeMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: metricsNamespace,
+		Subsystem: metricsSubsystem,
+		Buckets:   buckets,
+		Name:      "initial_build_or_pac_provision_creation_time",
+		Help:      "The time in seconds spent from the moment of Component creation till simple build pipeline submission, or PaC provision.",
+	})
```

Comment:
Would it make sense to split this one into two metrics, one for simple builds and one for PaC provisions? Otherwise we won't have the data about which one is slower

Reply:
hmm, ok

Reply:
`simpleBuildPipelineCreationTimeMetric` could be part of this metric. Creating a component is actually one way to trigger/request a simple build or PaC provision.

My understanding the consideration of this metric is it is just for the time counting for the first request upon Component creation. Whereas, both `simpleBuildPipelineCreationTimeMetric` and `pipelinesAsCodeComponentProvisionTimeMetric` collect the time for the first request (by creation) and subsequent requests.

However, if we need `simpleBuildPipelineCreationTimeMetric`, it would be good to split it even we have the other two. The overall time of PaC provision could be affected by more aspects (including the external APIs) than the simple build.

Reply:
So @tkdchen you're also +1 to have two separate onboarding metrics?

So that we would have
* simple component onboarding (component creation with a simple build request)
* PaC component onboarding (component creation with a PaC provision request)
* simple pipeline creation (any request for a simple build)
* PaC provisioning (any PaC provisioning request)

Reply:
Ack, that's fair.

Maybe the general metrics for simple builds and PaC provisions should exclude the onboarding cases then? Otherwise they could skew the metrics a bit by including these longer times

Reply:
I don't think that the onboarding cases will took a lot more, but
excluded them from those general metrics and counting it only for onboarding one